dbt/adapters/glue/impl.py

```diff
@@ -78,21 +78,18 @@ def convert_datetime_type(cls, agate_table, col_idx):
     def get_connection(self):
```

Comment:
as written in slack the issue in the current implementation is that `session` is not used in case of use_interactive_session_role_for_api_calls. You can consider to use something like this:


```
def get_connection(self):
    connection: GlueConnectionManager = self.connections.get_thread_connection()
    glueSession: GlueConnection = connection.handle
    if glueSession.credentials.role_arn is not None:
        if glueSession.credentials.use_interactive_session_role_for_api_calls is True:
            sts_client = boto3.client('sts')
            assumed_role_object = sts_client.assume_role(
                RoleArn=glueSession.credentials.role_arn,
                RoleSessionName="dbt"
            )
            credentials = assumed_role_object['Credentials']
            session = boto3.Session(
                aws_access_key_id=credentials['AccessKeyId'],
                aws_secret_access_key=credentials['SecretAccessKey'],
                aws_session_token=credentials['SessionToken']
            )
            glue_client = boto3.client("glue", region_name=glueSession.credentials.region,
                                  aws_access_key_id=credentials['AccessKeyId'],
                                  aws_secret_access_key=credentials['SecretAccessKey'],
                                  aws_session_token=credentials['SessionToken'])
            return glueSession, glue_client


    glue_client = boto3.client("glue", region_name=glueSession.credentials.region)

    return glueSession, glue_client
```
to keep supporting `use_interactive_session_role_for_api_calls` parameter

Reply:
Can you pass use_interactive_session_role_for_api_calls in the profiles.yml ? Shouldn't it be default to True instead of False ?

Reply:
Yes can be put in the profiles as it's pick from credentials.
Imho I will make default to True, as imply that the user maintains only one role